src/utils/SiriusAdapter.cpp

```diff
@@ -127,6 +147,62 @@ class TOPPSiriusAdapter :
     registerFlag_("no_recalibration", "If this option is set, SIRIUS will not recalibrate the spectrum during the analysis.", false);
   }
 
+  // extract adduct information from featureXML (MetaboliteAdductDecharger)
+  void extractAdductInformation(const PeakMap & spectra, const  String & adductinfo, map<size_t, StringList> & map_precursor_to_adducts)
```

Comment:
better to remove parameter adductinfo (=filename) and pass a const FeatureMap & instead. Load featureXML outside of method